tests/jballerina-unit-test/src/test/resources/test-src/statements/matchstmt/varbindingpatternmatchpattern/error_binding_pattern_negative2.bal

```diff
@@ -0,0 +1,59 @@
+// Copyright (c) 2021 WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+type FooError error<record {|string message?; string cause; boolean fatal?;string...; |}>;
+
+FooError err = error("Error One", message = "Msg One", cause = "Cause One", fatal = false);
+
+function testNonRequiredFieldBindingNegative() {
+    FooError error(reason1, message = message1, cause = cause1, fatal = fatal1) = err;
+    FooError error FooError(reason2, message = message2, cause = cause2, fatal = fatal2) = err;
+
+    error error(reason3, message = message3, cause = cause3, fatal = fatal3) = err;
+    error error FooError(reason4, message = message4, cause = cause4, fatal = fatal4) = err;
+
+    string? reason5;
+    string? reason6;
+    string? message5;
+    string? message6;
+    string? cause5;
+    string? cause6;
+    boolean? fatal5;
+    boolean? fatal6;
+
+    error(reason5, message = message5, cause = cause5, fatal = fatal5) = err;
+    error FooError(reason6, message = message6, cause = cause6, fatal = fatal6) = err;
+}
+
+function testUndefinedErrorDetailsNegative() {
+    FooError error(reason1, msg = msg1, cause = cause1, extra = extra1) = err;
```

Comment:
Can we add these test cases with `var` as well. looks like it does not give any error
```bal
var error(reason1, msg = msg1, cause = cause1, extra = extra1) = err; // no error
```

Reply:
Fixed with 9dc44c0